### PR TITLE
Issue #10773: suppress false scanner exception in xml meta reader

### PR DESCRIPTION
**Pull request by rnveach**
Issue #10773

Tested with eclipse-cs test area and it solved the issue.

### CODE REVIEW

**Review by romani — CHANGES_REQUESTED** [review]
Item:

File: src/main/java/com/puppycrawl/tools/checkstyle/meta/XmlMetaReader.java [hunk 1]
```diff
@@ -77,10 +78,17 @@ private XmlMetaReader() {
...
-            allMetadataSources.addAll(thirdPartyModuleFileNames);
+            try {
+                final Set<String> thirdPartyModuleFileNames =
+                        new Reflections(packageName, new ResourcesScanner())
+                                .getResources(Pattern.compile(".*checkstylemeta-.*\\.xml"));
+                allMetadataSources.addAll(thirdPartyModuleFileNames);
+            }
+            catch (ReflectionsException ex) {
+                if (!"Scanner ResourcesScanner was not configured".equals(ex.getMessage())) {
+                    throw ex;
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/meta/XmlMetaReader.java:89]: Can we do:
```
If (){
// Comment why we swallow exception
}else {throw ex;}
```

**Review by romani — CHANGES_REQUESTED** [review]
Item:

File: src/main/java/com/puppycrawl/tools/checkstyle/meta/XmlMetaReader.java (hunk 1 above)
Comment (rnveach) [reply src/main/java/com/puppycrawl/tools/checkstyle/meta/XmlMetaReader.java:89]:
````
XmlMetaReader:88 Rule:EmptyIfStmt Priority:3 Avoid empty if statements.
````

Comment (rnveach) [reply src/main/java/com/puppycrawl/tools/checkstyle/meta/XmlMetaReader.java:93]: See new commit.

Comment (romani) [reply src/main/java/com/puppycrawl/tools/checkstyle/meta/XmlMetaReader.java:93]: I am not sure why but I like easy to read code more than follow this Rule.
Let's see reaction on next rewiever on this.
If I am alone, ok to merge.

**Review by romani — APPROVED** [review]
Ok to merge.

File: pom.xml [hunk 2]
```diff
@@ -1075,12 +1075,12 @@
...
                       <counter>LINE</counter>
                       <value>COVEREDRATIO</value>
-                      <minimum>0.74</minimum>
+                      <minimum>0.71</minimum>
                     </limit>
                     <limit>
                       <counter>BRANCH</counter>
                       <value>COVEREDRATIO</value>
-                      <minimum>0.71</minimum>
+                      <minimum>0.66</minimum>
```
Comment (strkkk) [inline pom.xml:1083]: why we cannot add unit tests for this change?

File: src/main/java/com/puppycrawl/tools/checkstyle/meta/XmlMetaReader.java (hunk 1 above)
Comment (romani) [reply src/main/java/com/puppycrawl/tools/checkstyle/meta/XmlMetaReader.java:93]: @strkkk,  @pbludov , please share your opinion on this

Comment (pbludov) [reply src/main/java/com/puppycrawl/tools/checkstyle/meta/XmlMetaReader.java:93]: Maybe this comment should be extracted to a method with Javadoc? Something like
```
if (propagateResourceScannerException(ex)) {
     throw ex;
}
```

Comment (strkkk) [reply src/main/java/com/puppycrawl/tools/checkstyle/meta/XmlMetaReader.java:93]: I like solution proposed above

Comment (romani) [reply src/main/java/com/puppycrawl/tools/checkstyle/meta/XmlMetaReader.java:93]: @strkkk , do you mean solution from @pbludov ?

**Comment by rnveach** [comment]
@strkkk ping

File: src/main/java/com/puppycrawl/tools/checkstyle/meta/XmlMetaReader.java (hunk 1 above)
Comment (strkkk) [reply src/main/java/com/puppycrawl/tools/checkstyle/meta/XmlMetaReader.java:93]: yes

**Comment by strkkk** [comment]
@rnveach please take a look at comment https://github.com/checkstyle/checkstyle/pull/10786#discussion_r704071396
also there are  some CI failures

File: src/main/java/com/puppycrawl/tools/checkstyle/meta/XmlMetaReader.java (hunk 1 above)
Comment (rnveach) [reply src/main/java/com/puppycrawl/tools/checkstyle/meta/XmlMetaReader.java:93]: What is the best way to suppress this? Can it be suppressed by saying an if statement with a comment, do I suppress by method, or the full class? Let me know what the suppression is as I am not sure how to generate a tree for PMD.

Comment (rnveach) [reply src/main/java/com/puppycrawl/tools/checkstyle/meta/XmlMetaReader.java:93]: So which way am I changing the code to?

Comment (strkkk) [reply src/main/java/com/puppycrawl/tools/checkstyle/meta/XmlMetaReader.java:93]: I think this one https://github.com/checkstyle/checkstyle/pull/10786#discussion_r704071396 if @romani does not mind

Comment (romani) [reply src/main/java/com/puppycrawl/tools/checkstyle/meta/XmlMetaReader.java:93]: I am ok with Pavel proposal.

Comment (rnveach) [reply src/main/java/com/puppycrawl/tools/checkstyle/meta/XmlMetaReader.java:89]: @romani @strkkk @pbludov Done.

**Comment by rnveach** [comment]
@strkkk Done.

**Comment by strkkk** [comment]
@rnveach there are ci failures

**Comment by rnveach** [comment]
> [WARNING] Rule violated for class com.puppycrawl.tools.checkstyle.meta.XmlMetaReader: lines covered ratio is 0.70, but expected minimum is 0.71
[WARNING] Rule violated for class com.puppycrawl.tools.checkstyle.meta.XmlMetaReader: branches covered ratio is 0.62, but expected minimum is 0.66

Should be fixed now.

**Review by pbludov — APPROVED** [review]

**Comment by rnveach** [comment]
This was rebased due to conflicts.

@pbludov Are we ready to merge or will this go to a 3rd person?

**Review by strkkk — APPROVED** [review]

**Pull request merged**